Mathlib/CategoryTheory/Preadditive/Transfer.lean

```diff
@@ -0,0 +1,38 @@
+/-
+Copyright (c) 2025 Jakob von Raumer. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jakob von Raumer
+-/
+import Mathlib.CategoryTheory.Preadditive.AdditiveFunctor
+import Mathlib.Algebra.Equiv.TransferInstance
+
+/-!
+# Pulling back a preadditive structure along a fully faithful functor
+
+A preadditive structure on a category `D` transfers to a preadditive structure on `C` for a given
+fully faithful functor `F : C ⥤ D`.
+-/
+namespace CategoryTheory.Preadditive
+
+open Limits
+
+universe v₁ v₂ u₁ u₂
+
+variable {C : Type u₁} [Category.{v₁} C]
+variable {D : Type u₂} [Category.{v₂} D] [Preadditive D]
+variable {F : C ⥤ D} (hF : F.FullyFaithful)
+
+/-- If `D` is a preadditive category, any fully faithful functor `F : C ⥤ D` induces a preadditive
+structure on `C`. -/
+def ofFullyFaithful : Preadditive C where
+  homGroup P Q := hF.homEquiv.addCommGroup
+  add_comp P Q R f f' g := hF.map_injective (by simp [Equiv.add_def])
+  comp_add P Q R f g g' := hF.map_injective (by simp [Equiv.add_def])
+
+instance additive_ofFullyFaithful :
+    let _ : Preadditive C := ofFullyFaithful hF
```

Comment:
What's the best way to get around this? Can't make it a local instance because there's no synthetization order.

Reply:
Have you considered that perhaps this really should not be an instance and whenever I want to use `.ofFullyFaithful` then I should provide the `Additive` instance manually?

Reply:
The question whether `additive_ofFullyFaithful` should be an instance is orthogonal to the question about the `let`s needed, isn't it? Or did you mean the `Preadditive C` argument in `Functor.Additive` itself?

Reply:
Ah, I see. I think this is actually a fine use case for `letI`.

Reply:
Is there any way so that I don't have to repeat it? :confused:

Reply:
I think if there was a better way, then `Equiv.add_def` would use it.

Reply:
I meant the fact that I have to state that line in the signature as well as in the body of the instance :sweat_smile:

Reply:
Oh, sorry for being dense. I can't see a way to avoid that.